modules/social_features/social_content_block/src/ContentBuilder.php

```diff
@@ -36,109 +35,117 @@ class ContentBuilder implements ContentBuilderInterface {
+   *   The content block manager.
    */
   public function __construct(
     EntityTypeManagerInterface $entity_type_manager,
-    Connection $connection,
     ModuleHandlerInterface $module_handler,
-    TranslationInterface $string_translation
+    TranslationInterface $string_translation,
+    Connection $connection,
+    ContentBlockManagerInterface $content_block_manager
   ) {
     $this->entityTypeManager = $entity_type_manager;
-    $this->connection = $connection;
     $this->moduleHandler = $module_handler;
     $this->setStringTranslation($string_translation);
+    $this->connection = $connection;
+    $this->contentBlockManager = $content_block_manager;
   }
 
   /**
-   * Function to get all the topics based on the filters.
+   * Function to get all the entities based on the filters.
    *
-   * @param \Drupal\block_content\Entity\BlockContent $block_content
+   * @param \Drupal\block_content\BlockContentInterface $block_content
    *   The block content where we get the settings from.
    *
-   * @return array|string
-   *   Return the topics found.
+   * @return array
+   *   Returns the entities found.
    *
    * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
    * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
-  protected function getTopics(BlockContent $block_content) {
-    // Get topic type tags.
-    $topic_types_list = $block_content->get('field_topic_type')->getValue();
-    $topic_types = array_map(function ($topic_type) {
-      return $topic_type['target_id'];
-    }, $topic_types_list);
-
-    // Get group tags.
-    $group_tag_list = $block_content->get('field_group')->getValue();
-    $group_tags = array_map(function ($group_tag) {
-      return $group_tag['target_id'];
-    }, $group_tag_list);
-
-    // Get social tags.
-    $social_tag_list = $block_content->get('field_content_tags')->getValue();
-    $social_tags = array_map(function ($social_tag) {
-      return $social_tag['target_id'];
-    }, $social_tag_list);
-
-    // Use database select because we need joins
-    // which are not possible with entityQuery.
-    /** @var \Drupal\Core\Database\Query\SelectInterface $query */
-    $query = $this->connection->select('node_field_data', 'n')
-      ->fields('n', ['nid'])
-      ->condition('n.type', 'topic');
-
-    // Add topic type tags.
-    if (!empty($topic_types)) {
-      $query->innerJoin('node__field_topic_type', 'tt', 'tt.entity_id = n.nid');
-      $query->condition('tt.field_topic_type_target_id', $topic_types, 'IN');
+  protected function getEntities(BlockContentInterface $block_content) {
+    $plugin_id = $block_content->field_plugin_id->value;
+    $definition = $this->contentBlockManager->getDefinition($plugin_id);
+
+    if ($block_content->field_plugin_field->isEmpty()) {
+      $field_names = $definition['fields'];
+    }
+    else {
+      $field_names = [$block_content->field_plugin_field->value];
```

Comment:
Maybe some comments on what the difference is? When would the field stored on the content block differ from the default field set of the plugin? Can the user select a subset of fields?

Reply:
> Maybe some comments on what the difference is?

Done

> When would the field stored on the content block differ from the default field set of the plugin?

A stored field should be added to one from plugins

> Can the user select a subset of fields?

Yes, it's possible when select **All** option in the **Content selection** field